main.c

```diff
@@ -181,6 +188,90 @@ static void show_table_dump(ryzen_access ry, int any_adjust_applied)
 	//don't free current_table_values because this would deinitialize our table
 }
 
+// Check if launched from ryzen-controller
+static int check_parent() {
+#ifndef _WIN32
+	pid_t ppid = getppid();
+	char pathname[512];
+	char buf[512];
+	char gt[38] = "/opt/Ryzen Controller/ryzen-controller";
```

Comment:
Is it ok to check for such a specific path, or should we check only a part of it?

Reply:
It depends how much you want to limit access to the tool. Wanted it to be possible to limit argument access to a root controlled path. I.E it is possible to input values that can be quite bad for your system, A suid binary with access to /dev/mem is also a nice target for exploration. The harder to inject arbitrary data the better. I guess it is quite easy to call ryzenadj via `gdb -p` on  `ryzen-controller`.

If we only looked on part of the path it would be possible create a file named `ryzen-controller` with below 

`#!/bin/bash
ryzenadj "$@"
`

As said before. It all depends on how hard you want to limit the access.

Was thinking of adding a build option to disable use of /dev/mem and force use the DKMS SMU driver. I.E can not limit physical read access to just some addresses if /dev/mem is used.

The PCI config access can be limit access via apparmor. I.E can not write to random BAR's.